### ISSUE TITLE
[Help wanted] DETR (DEtection TRansformer): End-to-End Object Detection with Transformers

### ISSUE THREAD

**Issue by jaeyounkim** [tensorflow/models#8707]
# Help wanted: Research paper code and models

Please help us to implement the following research paper using TensorFlow 2.

If you want to contribute, please leave a comment to express your interest.

## Research papers

[![Paper](http://img.shields.io/badge/paper-arXiv.2005.12872-B3181B?logo=arXiv)](https://arxiv.org/abs/2005.12872)

* DETR (DEtection TRansformer) [End-to-End Object Detection with Transformers](https://arxiv.org/abs/2005.12872)

## Status

* Need contribution

### Contributor(s)

* Waiting for contributions

## Tasks

* Implement a paper in TensorFlow 2
* Reproduce the paper results
* Publish documentation
* Publish pre-trained SavedModel files at TensorFlow Hub

## Development environment

* Target
  * Repository: [Community directory](https://github.com/tensorflow/models/tree/master/community) or [Research directory](https://github.com/tensorflow/models/tree/master/research)
 * Branch: https://github.com/tensorflow/models/tree/detr

## Requirements

* Follow our [contribution guidelines](https://github.com/tensorflow/models/wiki/How-to-contribute)

## License

By contributing, you agree that your contributions will be licensed under Apache License 2.0.

**Comment by vk1996** [comment]
Interested

**Comment by SchernHe** [comment]
Also interested

**Comment by veds12** [comment]
Can I take this up?

**Comment by dresh** [comment]
I am interested too

**Comment by SchernHe** [comment]
Hey all, we already started implementing the DETR: https://github.com/auvisusAI/detr-tensorflow
There are still many open topics left (e.g. the additional mask-head), but maybe we can use our repo as a starting point?

General remark: We are a quite young start-up and this is our first community project. So we are happy for all kinds of feedback/remarks :)

**Comment by bhack** [comment]
Please take a look at https://github.com/tensorflow/community/pull/260 /cc @saberkun

**Comment by bhack** [comment]
Some small initial components are landing at https://github.com/tensorflow/models/pull/8962

**Comment by aavishkarmishra** [comment]
Is this sill open, I would like to work on this.

**Comment by hd1090** [comment]
Also Interested

**Comment by bhack** [comment]
Others PRs are landing:
https://github.com/tensorflow/models/pull/9130
https://github.com/tensorflow/models/pull/9135

**Comment by notacodingguy** [comment]
Interested

**Comment by thibo73800** [comment]
Here is a tensorflow implementation : https://github.com/Visual-Behavior/detr-tensorflow

**Comment by jaeyounkim** [comment]
We're currently developing this internally using. It will be released to the [beta projects directory](https://github.com/tensorflow/models/tree/master/official/vision/beta/projects) (ETA: End of Q2, 2021)

If you have a full implementation that meets our [guidelines](https://github.com/tensorflow/models/wiki/Research-paper-code-contribution), please let us know. We can consider adding it as a community model to the TensorFlow Model Garden.
